### PR TITLE
Only set the comment when persist_docs.relation is set

### LINKED ISSUE #343: dbt-trino adds comment into table create statement by default
### Expected behavior

dbt-core has persist_docs by default disabled. And dbt-trino should behave same as dbt-core.

https://docs.getdbt.com/reference/resource-configs/persist_docs

Additionally documentation is missing on official dbt documentation page. Trino is not listed on supported adapters list.

- https://docs.getdbt.com/reference/resource-configs/persist_docs
- https://docs.getdbt.com/reference/resource-configs/trino-configs

### Actual behavior

dbt-trino adds comment on table creation. When a description for a model exists on schema.yml

``` 
create table iceberg.<schema>.<table>
    comment ''
    as (
    SELECT
     *
    FROM
    table
    )
 ```

### Steps To Reproduce

- Do not set persist_docs config anywhere.
- Run a model with dbt trino where for a model description exists.

### Log output/Screenshots

_No response_

### Operating System

Windows 10

### dbt version

dbt-core==1.6.0, dbt-trino==1.6.1

### Trino Server version

422

### Python version

3.10

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by Fokko**
## Overview

I accidentally changed default behavior in https://github.com/starburstdata/dbt-trino/pull/317

Resolves #343

Checked that it only sets the comment when:

```json
{{ config(
    ...
    persist_docs={"relation": true}
) }}
```

And with the following configuration it won't set the comment:

```json
{{ config(
    ...
    persist_docs={"relation": false}
) }}
```

```json
{{ config(
    ...
) }}
```

## Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] `README.md` updated and added information about my change
- [x] I have run `changie new` to [create a changelog entry](https://github.com/starburstdata/dbt-trino/blob/master/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by Fokko** [comment]
Gentle ping @damian3031

File: dbt/include/trino/macros/adapters.sql [hunk 1]
```diff
@@ -89,8 +89,12 @@
 {%- endmacro -%}
 
 {% macro comment(comment) %}
-  {%- if comment is not none and comment|length > 0 -%}
-      comment '{{ comment | replace("'", "''") }}'
+  {%- set persist_docs = model['config'].get('persist_docs', {}) -%}
```
Comment (damian3031) [inline dbt/include/trino/macros/adapters.sql:92]:
```suggestion
  {%- set persist_docs = model['config'].get('persist_docs') -%}
```
Here, `get` returns `{}` by default if given key is missing, no need to specify it explicitly

File: dbt/include/trino/macros/adapters.sql [hunk 2]
```diff
@@ -89,8 +89,12 @@
...
+  {%- if persist_docs is not none -%}
```
Comment (damian3031) [inline dbt/include/trino/macros/adapters.sql:93]:
```suggestion
  {%- if persist_docs -%}
```
We should check if `persist_docs` evaluates to True. Currently it checks if it is not None
(`{}` will pass current condition, but it shouldn't)

**Review by damian3031 — CHANGES_REQUESTED** [review]

**Comment by damian3031** [comment]
@Fokko I added tests for this change

**Comment by damian3031** [comment]
Gentle ping @Fokko 🙂 Please adjust to (or comment on) the requested changes, and PR is good to merge

**Comment by Fokko** [comment]
@damian3031 Sorry for not replying earlier, thanks for adding tests, and the changes look good 👍🏻

**Comment by damian3031** [comment]
@Fokko thanks! Could you squash these 2 new commits into the first one? (this one [`2432b48` (#344)](https://github.com/starburstdata/dbt-trino/pull/344/commits/2432b48a9a56e309932f51434ed8a33e5c014df1))

**Review by damian3031 — APPROVED** [review]

**Comment by damian3031** [comment]
@Fokko thanks for your contribution!

**Comment by alaturqua** [comment]
@Fokko Thank you for fixing this.

When is the next release actually?

**Comment by damian3031** [comment]
@alaturqua It has just been released in 1.6.2!

**Comment by Fokko** [comment]
Thanks @damian3031 for shepherding this, and sorry @alaturqua for breaking this

**Pull request merged**